ui/hooks/gitprovider.ts

```diff
@@ -21,10 +21,15 @@ export function useIsAuthenticated() {
     isAuthenticated: error ? false : res?.valid,
     loading,
     error,
-    req: (provider: GitProvider) => {
-      const headers = makeHeaders(_.bind(getProviderToken, this, provider));
-      req(applicationsClient.ValidateProviderToken({ provider }, { headers }));
-    },
+    req: useCallback(
+      (provider: GitProvider) => {
+        const headers = makeHeaders(_.bind(getProviderToken, this, provider));
+        req(
+          applicationsClient.ValidateProviderToken({ provider }, { headers })
+        );
+      },
+      [makeHeaders, getProviderToken, applicationsClient.ValidateProviderToken]
```

Comment:
This is the change we made due to `check` changing too many times and causing infinite loops when working on `GitAuth`